metadata_backend/api/handlers/submission.py

```diff
@@ -300,15 +300,16 @@ async def put_submission_path(self, req: Request) -> Response:
         else:
             raise web.HTTPNotFound(reason=f"'{req.path}' does not exist")
 
-        submission[schema] = data
-        JSONValidator(submission, "submission").validate
-
         op = "add"
         if schema in submission:
             op = "replace"
         patch = [
             {"op": op, "path": f"/{schema}", "value": data},
         ]
+
+        submission[schema] = data
+        JSONValidator(submission, "submission").validate
```

Comment:
I don't really see how moving this changed anything. Did I miss something, could you explain, please?

Reply:
the `if schema in submission` checks if the `schema` is in the `submission`, and having the  `submission[schema] = data`  before the condition check means that the `schema` key will be added to the dict, thus making the op always default to `replace`.

by changing the lines we will not be adding the `schema` to the `submission` dict.

Reply:
simple example I could think of is:
```
smth = {"smth" : 1}
 
if "cool" in smth:
    print("cool")

smth["cool"] = [1, 2]
```
vs

```
smth = {"smth" : 1}
smth["cool"] = [1, 2]
if "cool" in smth:
    print("cool")
```

Reply:
I got it, thanks 😅